### PR TITLE
[SASS-7369] Create ZeroEmissionGoodsVehicle page

### PR DESCRIPTION
**Pull request by Jake-Raffe**

### CODE REVIEW

File: app/forms/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import models.common.{TaxYear, UserType}
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class ZeroEmissionGoodsVehicleFormProvider @Inject() extends Mappings {
+
+  def apply(userType: UserType, taxYear: TaxYear): Form[Boolean] =
+    Form(
+      "value" -> boolean(s"zeroEmissionGoodsVehicle.error.required.$userType", args = Seq((taxYear.value - 1).toString, taxYear.value.toString))
```
Comment (lukastymo) [inline app/forms/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleFormProvider.scala:29]: small improvement suggestion: it's not immediately obvious what is `taxYear.value -1` and we use it quite often.

We can add a helper method inside TaxYear somethlig like:
```
final case class TaxYear(value: int) extends AnyVal {
  val startOfTheYear: Int = value - 1
}
```

File: app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              updatedUserAnswers                <- service.persistAnswer(businessId, editedUserAnswers, answer, ZeroEmissionGoodsVehiclePage)
+            } yield Redirect(navigator.nextPage(ZeroEmissionGoodsVehiclePage, redirectMode, updatedUserAnswers, taxYear, businessId))
+        )
+  }
+
+  private def handleGatewayQuestion(currentAnswer: Boolean,
+                                    request: DataRequest[_],
+                                    mode: Mode,
+                                    businessId: BusinessId): Future[(UserAnswers, Mode)] = {
+    val pagesToBeCleared: List[Settable[_]] = List()
```
Comment (lukastymo) [inline app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala:81]: More idiomatic way to create empty list is `= Nil`

**Review by lukastymo — APPROVED** [review]

File: app/forms/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleFormProvider.scala (hunk 1 above)
Comment (Jake-Raffe) [reply app/forms/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleFormProvider.scala:29]: Good idea, will do

File: app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala (hunk 2 above)
Comment (Jake-Raffe) [reply app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala:81]: Leaving an empty list to remember to add each page into the list as each page is created

Comment (lukastymo) [reply app/controllers/journeys/capitalallowances/zeroEmissionGoodsVehicle/ZeroEmissionGoodsVehicleController.scala:81]: ok

**Review by lukastymo — DISMISSED** [review]

**Review by lukastymo — APPROVED** [review]

**Pull request merged**